icu4j/main/classes/core/src/com/ibm/icu/impl/ICUResourceBundle.java

```diff
@@ -1481,10 +1527,24 @@ protected static ICUResourceBundle getAliasedResource(
             UResourceBundle requested) {
         WholeBundle wholeBundle = base.wholeBundle;
         ClassLoader loaderToUse = wholeBundle.loader;
+        String rpath = wholeBundle.reader.getAlias(_resource);
+        String baseName = wholeBundle.baseName;
+
+        int baseDepth = base.getResDepth();
+        String[] baseKeyPath = new String[baseDepth + 1];
+        base.getResPathKeys(baseKeyPath, baseDepth);
```

Comment:
Question: Is it ok to ask for one more path key than the base has?

Reply:
You mean pass `baseDepth + 1` to `getResPathKeys()`? No, that'll produce an `ArrayIndexOutOfBoundsException`.  I think the original code was appending a new item onto the end of the path in much the same way as I'm doing now.